helper_crates/vtable/tests/test_vrc.rs

```diff
@@ -150,3 +151,71 @@ fn rc_test_threading() {
     drop(weak);
     h.join().unwrap();
 }
+
+#[vtable]
+struct AppVTable {
+    drop_in_place: fn(VRefMut<AppVTable>) -> Layout,
+    dealloc: fn(&AppVTable, ptr: *mut u8, layout: Layout),
+}
+
+#[derive(Debug, const_field_offset::FieldOffsets)]
+#[repr(C)]
+#[pin]
+struct AppStruct {
+    some: SomeStruct,
+    another_struct: SomeStruct,
+}
+
+impl AppStruct {
+    fn new() -> VRc<AppVTable, Self> {
+        let string = Rc::new("hello".to_string());
+        let self_ = Self {
+            some: SomeStruct { e: 55, x: "_".into(), foo: string.clone() },
+            another_struct: SomeStruct { e: 100, x: "_".into(), foo: string.clone() },
+        };
+        VRc::new(self_)
+    }
+}
+
+impl App for AppStruct {}
+
+AppVTable_static!(static APP_STRUCT_TYPE for AppStruct);
+
+#[test]
+fn rc_map_test() {
+    fn get_struct_value(instance: &VRcMapped<AppVTable, SomeStruct>) -> u8 {
+        let field_ref = SomeStruct::FIELD_OFFSETS.e.apply_pin(instance.as_pin_ref());
+        *field_ref
+    }
+
+    let app_rc = AppStruct::new();
+
+    let some_struct_ref =
+        VRcMapped::map(&app_rc, |app| AppStruct::FIELD_OFFSETS.some.apply_pin(app));
+    let other_struct_ref =
+        VRcMapped::map(&app_rc, |app| AppStruct::FIELD_OFFSETS.another_struct.apply_pin(app));
```

Comment:
Looking at this several months after we initially wrote it, I still like the API and concept :-). It's kind of neat.